mkdocs/themes/mkdocs/toc.html

```diff
@@ -5,15 +5,18 @@
         </button>
     </div>
 
+    {% macro toc_item(item) %}
+        {%- if item.level <= config.theme.navigation_depth %}
+            <li class="nav-item" data-level="{{ item.level }}"><a href="{{ item.url }}" class="nav-link">{{ item.title }}</a></li>
+            {%- for child in item.children %}
+                {{- toc_item(child) }}
+            {%- endfor %}
+        {%- endif %}
+    {%- endmacro %}
```

Comment:
Currently, I kept this list non-hierarchical (i.e. no nested `<ul>`s) just because that's how it used to be. Maybe it'd be better to use nested `<ul>`s instead since that's what the TOC extension does for its HTML. Since I'm changing the classes for the `<li>`s, it's a compatibility break anyway (e.g. for `mkdocs-bootswatch`), so themes that inherit from this one will need an update no matter what...

Reply:
As this is theme specific and I'm not aware of any themes which inherit from the `mkdocs` theme except for `mkdocs-bootswatch`, backward incompatible changes in the HTML are not a big worry for me. If we are going to do it at all, it probably makes sense to break compatibility now rather than later.

As far as the proposed change, I don't really have a preference one way or the other.

Reply:
Ok, I went with nested `<ul>`s. I'll update the Bootswatch themes with the CSS changes here as well...